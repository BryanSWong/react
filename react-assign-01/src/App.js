import React, { Component } from 'react';
import './App.css';
import UserInput from './UserInput/UserInput';
import UserOutput from './UserOutput/UserOutput';

class App extends Component {
    state = {
        output: [
            {name:"Thisguy"},
            {name: "Thatguy"},
            {name: "Otherguy"}
        ]
    };

    changeNameHandler = () => {
        this.setState({output: [
            {name:"Thisguyoverthere"},
            {name: "Thatguy"},
            {name: "Otherguy"}
        ]})
    };

    newNameHandler = (event) => {
        this.setState({output: [
            {name: "Thisguyoverthere"},
            {name: event.target.value},
            {name: "Otherguy"}
        ]})
    };


    render() {
        const style = {
            backgroundColor: 'blue',
            font: 'inherit',
            margin: '2px solid green',
            padding: '10px',
            cursor: 'pointer',
            borderRadius: '12px'
        };

        return (
          <div className="App">
              <button style={style} onClick={this.changeNameHandler}>Change</button>
              <UserOutput name={this.state.output[0].name}/>
              <UserOutput name={this.state.output[1].name}/>
              <UserOutput name={this.state.output[2].name}/>
              <UserInput name={this.state.output[1].name} changed={this.newNameHandler}/>
          </div>
    );
  }
}

export default App;
