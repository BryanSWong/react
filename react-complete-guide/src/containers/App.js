import React, { Component } from 'react';
import classes from './App.css';
import Persons from '../Components/Persons/Persons';
import Cockpit from '../Components/Cockpit/Cockpit';


class App extends Component {

    constructor(props) {
        super(props);
        console.log('[App.js] inside constructor', props);
        this.state = {
            persons: [
                {id: 'yolds', name: 'Dude', age: 33 },
                {id: 'foldsnf', name: 'SomeGuy', age: 34 },
                {id: 'nbdsaof', name: 'ThisGuy', age: 35 }
            ],
            otherState: 'some other value',
            showPersons: false
        };
    }

    componentWillMount () {
        console.log('[App.js] Inside componentWillMount()');
    }

    componentDidMount(){
        console.log('[App.js] Inside componentDidMount()');
    }

    // state = {
    //     persons: [
    //         {id: 'yolds', name: 'Dude', age: 33 },
    //         {id: 'foldsnf', name: 'SomeGuy', age: 34 },
    //         {id: 'nbdsaof', name: 'ThisGuy', age: 35 }
    //     ],
    //     otherState: 'some other value',
    //     showPersons: false
    // };

    nameChangeHandler = (event, id) => {
        const personIndex = this.state.persons.findIndex(p => {
            return p.id === id
        });
        const person = {...this.state.persons[personIndex]};
        //const person = Object.assign({}, this.state.persons[personIndex]);
        person.name = event.target.value;
        const persons = [...this.state.persons];
        persons[personIndex] = person;
        this.setState({persons: persons});
    };


    togglePersonHandler = () => {
        const doesShow = this.state.showPersons;
        this.setState({showPersons: !doesShow});
    };

    deletePersonHandler = (personIndex) => {
        //const persons = this.state.persons.slice();
        const persons = [...this.state.persons]; // modern syntext
        persons.splice(personIndex, 1);
        this.setState({persons: persons})
    };


    render() {

        console.log('[App.js] Inside Render()');

        let persons = null;

        if(this.state.showPersons) {
            persons = <Persons
                persons = {this.state.persons}
                clicked = {this.deletePersonHandler}
                changed = {this.nameChangeHandler}/>;
        }

        return (

            <div className={classes.App}>
                <Cockpit
                    appTitle={this.props.title}
                    showPersons={this.state.showPersons}
                    persons={this.state.persons}
                    clicked={this.togglePersonHandler}/>
                {persons}
            </div>
            );

      // return React.createElement('div', {className: 'App'}, React.createElement('h1', null, 'Does this work now?'));
    }
}

export default App;
