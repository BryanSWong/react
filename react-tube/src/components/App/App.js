import React, { Component } from 'react';
import {Route, Switch} from 'react-router-dom';

import logo from '../../assets/images/logo.png';

import classes from './App.css';

import Login from '../Login/Login';

class App extends Component {

  render() {
    return (
      <div className={classes.App}>
        <header className={classes.AppHeader}>
          <img id="logo" src={logo} className={classes.AppLogo} alt="logo" />
          <h1 className={classes.AppTitle}>Welcome to React Tube</h1>
        </header>
        <Login />
      </div>
    );
  }
}

export default App;