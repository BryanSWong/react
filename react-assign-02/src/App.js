import React, { Component } from 'react';
import './App.css';
import Input from './Input/Input';
import Validation from './Validation/Validation';
import Char from './Char/Char';

class App extends Component {

    state = {
        input:[
            {word: ''}
        ]
    };

    inputHandler = (event) => {
        this.setState({input:[{word: event.target.value}]})
    };

    deleteHandler = (inputIndex) => {
        const input = [...this.state.input];
        input.splice(inputIndex, 1);
        this.setState({input: input})
    };

    render() {

        let input = null;

        if(this.state.input[0].word.length > 0){
            input = (
                <div>
                    {this.state.input[0].word.split('').map((text, index) => {
                        return <Char
                            letter={text}
                            key="123"
                            clicked={() => this.deleteHandler(index)}
                        />
                    })}
                </div>
            );
        }

        return <div className="App">
            <Input word={this.state.input[0].word} changed={this.inputHandler}/>
            <Validation word={this.state.input[0].word.length}/>
            {input}
         </div>
    }
}

export default App;


/*
* 1. Create input field (in app component) with a change listener which outputs the length
* of the text below it (e.g in a paragraph).
* COMPLETE
*
* 2. Create a new component (=> ValidationComponent) which receives the text length as a prop.
* COMPLETE
*
* 3. Inside the ValidationComponent, either output "text too short" or "Text long enough"
* depending on the ext length (e.g take 5 as a minimum length)
* COMPLETE
*
* 4. Create another component (=> CharComponent) and style it as inline box
* (=> display: inline-block, padding: 16 px, text-align: center, margin: 16px, border:1 px solid black
* COMPLETE
*
* 5. Render a list of CharComponents where each CharComponent receives a different letter entered text
* (in the initial input field) as a prop.
* COMPLETE
*
* 6. when you click a char component it should be removed form the entered text.
*
*
* */