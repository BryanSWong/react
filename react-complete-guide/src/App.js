import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Person from './Person/Person';


class App extends Component {
    state = {
        persons: [
            {name: 'Dude', age: 33 },
            {name: 'SomeGuy', age: 34 },
            {name: 'ThisGuy', age: 35 }
        ],
        otherState: 'some other value',
        showPersons: false
    };

    switchNameHandler = (newName) => {
        // console.log('was clicked');
        // DONT DO THIS this.state.persons[0].name = 'ThatGuy';
        this.setState({
            persons: [
                {name: newName, age: 37 },
                {name: 'SomeGuy', age: 34 },
                {name: 'ThisGuy', age: 35 }
            ]
        });
    };

    nameChangeHandler = (event) => {
        this.setState({
            persons: [
                {name: 'Dude', age: 37 },
                {name: event.target.value, age: 34 },
                {name: 'ThisGuy', age: 35 }
            ]
        });
    };

    togglePersonHandler = () => {
        const doesShow = this.state.showPersons;
        this.setState({showPersons: !doesShow});
    };

  render() {

        const style = {
            backgroundColor: 'white',
            font: 'inherit',
            border: '1px solid blue',
            padding: '8px',
            cursor: 'pointer'
        };

        let persons = null;

        if(this.state.showPersons){
            persons = (<div>
                <Person name={this.state.persons[0].name} age={this.state.persons[0].age} click={this.switchNameHandler.bind(this, 'Dude!')}>Hobbies: running</Person>
                <Person name={this.state.persons[1].name} age={this.state.persons[1].age} changed={this.nameChangeHandler}/>
                <Person name={this.state.persons[2].name} age={this.state.persons[2].age}/>
            </div>);
        }

    return (
      <div className="App">
          <h1>Hi Im a react app</h1>
          <p>This is really working</p>
          <button style= {style} onClick={this.togglePersonHandler}>Toggle Persons</button>
          {persons}
      </div>
    );

      // return React.createElement('div', {className: 'App'}, React.createElement('h1', null, 'Does this work now?'));
  }
}

export default App;
