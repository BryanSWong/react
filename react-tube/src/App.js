import React, { Component } from 'react';
import logo from './assets/images/logo.png';
import './App.css';

import Login from './components/Login/Login'

class App extends Component {

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} id="logo" className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React Tube</h1>
        </header>

          <Login/>

      </div>
    );
  }
}

export default App;
