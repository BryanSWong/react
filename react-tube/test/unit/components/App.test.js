import React from 'react';

import App from '../../../src/components/App/App';

describe('App Component', () => {

    const component = shallow(<App debug />);

    it('should render correctly in "debug" mode', () => {
        expect(component).toMatchSnapshot();
    });

});