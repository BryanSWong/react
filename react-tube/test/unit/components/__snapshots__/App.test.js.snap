// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly in "debug" mode 1`] = `
<div>
  <header>
    <img
      alt="logo"
      id="logo"
      src="logo.png"
    />
    <h1>
      Welcome to React Tube
    </h1>
  </header>
  <Login />
</div>
`;
